drivers/storage/portworx/component/storage_class.go

```diff
@@ -0,0 +1,198 @@
+	pxReplicatedStorageClass               = "px-replicated"
+	pxDbLocalSnapshotStorageClass          = "px-db-local-snapshot"
+	pxDbCloudSnapshotStorageClass          = "px-db-cloud-snapshot"
+	pxDbEncryptedStorageClass              = "px-db-encrypted"
+	pxReplicatedEncryptedStorageClass      = "px-replicated-encrypted"
+	pxDbLocalSnapshotEncryptedStorageClass = "px-db-local-snapshot-encrypted"
+	pxDbCloudSnapshotEncryptedStorageClass = "px-db-cloud-snapshot-encrypted"
+	portworxProvisioner                    = "kubernetes.io/portworx-volume"
+)
+
+type portworxStorageClass struct {
+	k8sClient client.Client
+}
+
+func (c *portworxStorageClass) Initialize(
+	k8sClient client.Client,
+	_ version.Version,
+	_ *runtime.Scheme,
+	_ record.EventRecorder,
+) {
+	c.k8sClient = k8sClient
+}
+
+func (c *portworxStorageClass) IsEnabled(cluster *corev1alpha1.StorageCluster) bool {
+	return true
+}
+
+func (c *portworxStorageClass) Reconcile(cluster *corev1alpha1.StorageCluster) error {
+	ownerRef := metav1.NewControllerRef(cluster, pxutil.StorageClusterKind())
+	docAnnotations := map[string]string{
+		"params/docs":              "https://docs.portworx.com/scheduler/kubernetes/dynamic-provisioning.html",
+		"params/fs":                "Filesystem to be laid out: none|xfs|ext4",
+		"params/block_size":        "Block size",
+		"params/repl":              "Replication factor for the volume: 1|2|3",
+		"params/secure":            "Flag to create an encrypted volume: true|false",
+		"params/shared":            "Flag to create a globally shared namespace volume which can be used by multiple pods: true|false",
+		"params/priority_io":       "IO Priority: low|medium|high",
+		"params/io_profile":        "IO Profile can be used to override the I/O algorithm Portworx uses for the volumes: db|sequential|random|cms",
+		"params/aggregation_level": "Specifies the number of replication sets the volume can be aggregated from",
+		"params/sticky":            "Flag to create sticky volumes that cannot be deleted until the flag is disabled",
+		"params/journal":           "Flag to indicate if you want to use journal device for the volume's metadata. This will use the journal device that you used when installing Portworx. It is recommended to use a journal device to absorb PX metadata writes",
+	}
+
+	storageClasses := []*storagev1.StorageClass{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:            pxDbStorageClass,
+				OwnerReferences: []metav1.OwnerReference{*ownerRef},
+				Annotations:     docAnnotations,
+			},
+			Provisioner: portworxProvisioner,
+			Parameters: map[string]string{
+				api.SpecHaLevel:   "3",
+				api.SpecIoProfile: "db",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:            pxDbEncryptedStorageClass,
+				OwnerReferences: []metav1.OwnerReference{*ownerRef},
+				Annotations: map[string]string{
+					"params/note": "Ensure that you have a cluster-wide secret created in the configured secrets provider",
```

Comment:
why are there no docAnnotations to the rest of the storage classes ?

Reply:
We could even add only the relevant annotations to each storage class, instead of putting all of them

Reply:
The idea was to have a basic storage class which talks about all the supported fields and also points to our docs website. We have also added a note on the encrypted storage class about how using cluster wide secret. Adding to all storage classes will make it very verbose, especially when they cannot change any of these.